crates/p2p/src/main_loop.rs

```diff
@@ -171,6 +178,49 @@ impl MainLoop {
             SwarmEvent::ConnectionEstablished {
                 peer_id, endpoint, ..
             } => {
+                // If this is an incoming connection, we have to prevent the peer from
+                // reconnecting too quickly.
+                if endpoint.is_listener() {
+                    // Extract the IP address of the peer from his multiaddr.
+                    let ip = endpoint.get_remote_address().iter().find_map(|p| match p {
```

Comment:
I wonder how we should treat the case of circuit relay connections here. If we have peers A, B, and relay R, then both A and B see the other peer as `/ip_/address_of_R/.../p2p-circuit/B|A` respectively.

Pathfinder upon connecting to a bootstrap node also [starts a relay listener via that node](https://github.com/eqlabs/pathfinder/blob/cebbda4057a940032b08dd4bc580af56444d502d/crates/pathfinder/src/p2p_network.rs#L80).